tests/regression/test_interpolate_cross_mesh.py

```diff
@@ -0,0 +1,243 @@
+        m_dest = UnitSquareMesh(3, 5, quadrilateral=True)
+        coords = np.array(
+            [[0.56, 0.6], [0.1, 0.9], [0.9, 0.1], [0.9, 0.9], [0.726, 0.6584]]
+        )  # fairly arbitrary
+        # add the coordinates of the mesh vertices to test boundaries
+        vertices_src = allgather(m_src.comm, m_src.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_src))
+        vertices_dest = allgather(m_dest.comm, m_dest.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_dest))
+        expr_op = mul
+        V_src = FunctionSpace(m_src, "CG", 3)
+        V_dest = FunctionSpace(m_dest, "CG", 4)
+        V_dest_2 = FunctionSpace(m_dest, "DG", 2)
+    elif request.param == "circlemanifold":
+        m_src = UnitSquareMesh(2, 3)
+        # shift to cover the whole circle
+        m_src.coordinates.dat.data[:] *= 2
+        m_src.coordinates.dat.data[:] -= 1
+        m_dest = CircleManifoldMesh(1000)  # want close to actual unit circle
+        coords = np.array(
+            [
+                [0, 1],
+                [1, 0],
+                [-1, 0],
+                [0, -1],
+                [sqrt(2) / 2, sqrt(2) / 2],
+                [-sqrt(3) / 2, 1 / 2],
+            ]
+        )  # points that ought to be on the unit circle
+        # only add target mesh vertices since they are common to both meshes
+        vertices_dest = allgather(m_dest.comm, m_dest.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_dest))
+        # function.at often gets conflicting answers across boundaries for this
+        # mesh, so we lower the tolerance a bit for this test
+        m_dest.tolerance = 0.5
+        expr_op = mul
+        V_src = FunctionSpace(m_src, "CG", 3)
+        V_dest = FunctionSpace(m_dest, "CG", 4)
+        V_dest_2 = FunctionSpace(m_dest, "DG", 2)
+    elif request.param == "extrudedcube":
+        m_src = ExtrudedMesh(UnitSquareMesh(2, 3), 2)
+        m_dest = UnitCubeMesh(3, 5, 7)
+        coords = np.array(
+            [
+                [0.56, 0.06, 0.6],
+                [0.726, 0.6584, 0.951],
+            ]
+        )  # fairly arbitrary
+        vertices_src = allgather(m_src.comm, m_src.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_src))
+        vertices_dest = allgather(m_dest.comm, m_dest.coordinates.dat.data_ro)
+        coords = np.concatenate((coords, vertices_dest))
+        # we use add to avoid TSFC complaints about too many indices for sum
+        # factorisation
+        expr_op = add
+        V_src = FunctionSpace(m_src, "CG", 2)
+        V_dest = FunctionSpace(m_dest, "CG", 3)
+        V_dest_2 = FunctionSpace(m_dest, "CG", 1)
+    else:
+        raise ValueError("Unknown mesh pair")
+    expr = reduce(expr_op, SpatialCoordinate(m_src))
+    expected = reduce(expr_op, coords.T)
+    return m_src, m_dest, coords, expr, expected, V_src, V_dest, V_dest_2
+
+
+def test_interpolate_cross_mesh(parameters):
```

Comment:
I need to add tests where we have exact refinements of meshes to check for zero error in the interpolation and transpose interpolation operations